xla/service/gpu/gpu_compiler.cc

```diff
@@ -884,6 +868,44 @@ absl::Status RunCollectiveOptimizationPasses(
         /*reuse_pipelined_op_buffer=*/HloPredicateFalse};
     collectives_pipeline.AddPass<CollectivePipeliner>(config);
   }
+  return absl::OkStatus();
```

Comment:
line 924 AddP2PPipeliner(collectives_pipeline);
also invokes CP and can potentially cause the same problem, and is not protected by you flag. 
Can you also include this?

Reply:
Would the p2p pipeliner have any dependency on other pipeliner passes? if not i'd suggest to not move it for now, the purpose for this is to not break the QDQ patterns of fp8 quantization, i'm not sure if that breakage will ever happen with pipeline parallelism since we likely wont shard the graph in the middle of a QDQ.